Maple2.Server.Game/Manager/BeautyManager.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Linq;
+using Maple2.Database.Storage;
+using Maple2.Model.Enum;
+using Maple2.Model.Error;
+using Maple2.Model.Game;
+using Maple2.Model.Metadata;
+using Maple2.Server.Game.Manager.Items;
+using Maple2.Server.Game.Packets;
+using Maple2.Server.Game.Session;
+using Serilog;
+
+namespace Maple2.Server.Game.Manager;
+
+public sealed class BeautyManager : IDisposable {
+    private readonly GameSession session;
+    private readonly ItemCollection items;
+    private Item? previousHair;
+    public BeautyManager(GameSession session) {
+        this.session = session;
+        items = new ItemCollection((short) (Constant.BaseStorageCount + session.Player.Value.Unlock.HairSlotExpand));
+        using GameStorage.Request db = session.GameStorage.Context();
+        foreach (Item item in db.GetSavedHairs(session.CharacterId).Where(item => items.Add(item).Count == 0)) {
+            Log.Error("Failed to add saved hair:{Uid}", item.Uid);
+        }
+    }
+
+    public void Dispose() {
+        using GameStorage.Request db = session.GameStorage.Context();
+        lock (session.Item) {
+            db.SaveItems(session.CharacterId, items.ToArray());
+        }
+    }
+
+    public void Load() {
+            session.Send(BeautyPacket.StartList());
+            session.Send(BeautyPacket.SaveSlots(session.Player.Value.Unlock.HairSlotExpand));
+            session.Send(BeautyPacket.ListCount(items.Count));
+            if (items.Count > 0) {
+                session.Send(BeautyPacket.ListHair(items.OrderBy(hair => hair.CreationTime).ToList()));
+                ;
+            }
+    }
+
+    public void AddHair(long uid) {
+        Item? hair = session.Item.GetOutfit(uid);
+        if (hair == null || !hair.Type.IsHair) {
+            return;
+        }
+
+        if (items.Contains(uid)) {
+            return;
+        }
+
+        if (items.OpenSlots <= 0) {
+            return;
+        }
+
+        Item? hairCopy = hair.Clone();
```

Comment:
~~`Clone()` keeps the [same uid](https://github.com/kOchirasu/Maple2/blob/241807a62048bacd1b3422e4bd4294735d533b9c/Maple2.Model/Game/Item/Item.cs#L96) so I'm not sure if this is doing what you want. You can double check what happening (maybe it gives you a new uid when you call `CreateItem()`.~~

Okay so I double checked and that is the case. Please add a comment explaining this behavior so it's not confusing:
> Clone() causes hairCopy has the same Uid as hair. However, CreateItem() will force a new Uid.